### PR TITLE
Add support for quay.io registry

### PR DESCRIPTION
**Pull request by phracek**
This commit adds support for quay.io registry.

The PR https://github.com/sclorg/container-common-scripts/pull/181
has to be merged before and common directory has to be updated as well.

Signed-off-by: Petr "Stone" Hracek <phracek@redhat.com>

### CODE REVIEW

**Comment by phracek** [comment]
Quay.io registry will be used for CentOS7 images instead of dockerhub.io:
* https://quay.io/repository/centos7/s2i-base-centos7
* https://quay.io/repository/centos7/s2i-core-centos7

**Comment by pkubatrh** [comment]
```
16:28:01 Trying to pull repository docker.io/library/centos ... 
16:28:01 toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
```

Yeah, that is what I was afraid of :/ I wonder if we can take the centos bae image from somewhere else (centos registry? Quay?)

**Comment by phracek** [comment]
> ```
> 16:28:01 Trying to pull repository docker.io/library/centos ... 
> 16:28:01 toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
> ```
> 
> Yeah, that is what I was afraid of :/ I wonder if we can take the centos base image from somewhere else (centos registry? Quay?)
> 
> Edit: `registry.centos.org/centos/centos:7` might be a good candidate

Sorry, but `quay.io` was not used yet. Still waiting for merge https://github.com/sclorg/container-common-scripts/pull/181.
As soon as this is merged, I will update `common` directory as well.

**Comment by phracek** [comment]
Any reason why not `quay.io`?

**Comment by phracek** [comment]
Or we can use `quay.io/generic/centos7`. Unfortunately, `quay.io` does not provide any information about this registry. I would prefer  to stay in registry.centos.org

**Comment by phracek** [comment]
All s2i-{php,ruby,python,perl}-container are built when s2i-base-container is changed. But no PR is filed for CentOS7 quay.io for them. Let's see what happened with tests.

**Comment by phracek** [comment]
`centos7` test failed on `conu`.

File: base/test/run [hunk 1]
```diff
@@ -6,7 +6,7 @@
 # IMAGE_NAME specifies a name of the candidate image used for testing.
 # The image has to be available before this script is executed.
 #
-IMAGE_NAME=${IMAGE_NAME-centos/s2i-base-centos7-candidate}
+IMAGE_NAME=${IMAGE_NAME-${REGISTRY}centos7/s2i-base-centos7-candidate}
```
Comment (pkubatrh) [inline base/test/run:9]: I would take the opportunity to remove the default altogether:

```suggestion
IMAGE_NAME=${IMAGE_NAME}
```

Comment (pkubatrh) [reply base/test/run:9]: Or maybe add a new check to see if the variable is set: 
```suggestion
test -n "${IMAGE_NAME-}" || { false 'make sure $IMAGE_NAME is defined'; exit 1 }
```

File: core/test/run [hunk 2]
```diff
@@ -6,7 +6,7 @@
 # IMAGE_NAME specifies a name of the candidate image used for testing.
 # The image has to be available before this script is executed.
 #
-IMAGE_NAME=${IMAGE_NAME-centos/s2i-core-centos7-candidate}
+IMAGE_NAME=${IMAGE_NAME-${REGISTRY}centos7/s2i-core-centos7-candidate}
```
Comment (pkubatrh) [inline core/test/run:9]: Same here

Comment (pkubatrh) [inline core/test/run]:
```suggestion
test -n "${IMAGE_NAME-}" || { echo 'make sure $IMAGE_NAME is defined'; exit 1; }
```

**Comment by phracek** [comment]
CentOS 7 tests failed on `conu` in s2i-ruby-container.

**Pull request merged**